### ISSUE TITLE
DatePicker: unwanted text selection when clicking through months

### ISSUE THREAD

**Issue by TheophileMot** [carbon-design-system/carbon-components-svelte#993]
If you click the "back/forward" arrows too quickly in a DatePicker calendar, the browser will select text from the calendar. Specifically, this happens when the clicking is fast enough to be a double-click.

To reproduce: 

1. Open a [DatePicker](https://carbon-components-svelte.onrender.com/components/DatePicker) with a calendar (e.g., the "Single" mode)
2. Click one of the arrows quickly. This will select the first day of the week (right arrow), or the name of the month (left arrow).

![image](https://user-images.githubusercontent.com/34871447/148852689-28491a7e-7157-403a-a34a-24763bd4d8eb.png)

**Comment by metonym** [comment]
Nice catch – this also appears to affect the [canonical React implementation](https://react.carbondesignsystem.com/?path=/story/components-datepicker--single).

**Comment by TheophileMot** [comment]
@metonym Indeed, it looks like it stems from the underlying [flatpickr](https://flatpickr.js.org/examples/) library.

**Comment by metonym** [comment]
@TheophileMot I've raised the issue in the `flatpickr` and `carbon` repos.

Meanwhile, you apply the following styles as a workaround:

```svelte
<style>
  :global(.flatpickr-prev-month, .flatpickr-next-month) {
    user-select: none;
  }
</style>
```

**Comment by metonym** [comment]
Fixed in [v0.58.0](https://github.com/carbon-design-system/carbon-components-svelte/releases/tag/v0.58.0).

v0.58.0 uses `carbon-components` 10.52.0, which contains a [fix for preventing unwanted text selection](https://github.com/carbon-design-system/carbon/issues/10376).
